.gitlab-ci.yml

```diff
@@ -51,38 +55,94 @@ build_doc:
     paths:
       - doc/build
 
+build_suse_doc:
+  image: $CI_REGISTRY/$BUILD_IMAGES_PROJECT:$TUMBLEWEED_BUILD
+  stage: documentation
+  script:
+    - tox -e doc_suse
+  artifacts:
+    paths:
+      - doc/build/
+
+push_suse_doc:
+  image: $CI_REGISTRY/$BUILD_IMAGES_PROJECT:$TUMBLEWEED_BUILD
+  stage: documentation
+  only:
+    - master
+  script:
+    - "export GIT_SSH=$(pwd)/run_ssh"
+    - "export ID_DOC_SUSE=$(pwd)/id_doc_suse"
+    - echo 'exec ssh -o StrictHostKeyChecking=no -i $ID_DOC_SUSE "$@"' > run_ssh
+    - chmod u+x run_ssh
+    - base64 --decode "$DOC_SUSE_SECRET" > id_doc_suse
+    - chmod 600 id_doc_suse
+    - git clone git@github.com:OSInside/kiwi-suse-doc.git /tmp/kiwi-suse-doc
+    - cp doc/DC-kiwi /tmp/kiwi-suse-doc/doc/build
+    - cp doc/source/.images/* /tmp/kiwi-suse-doc/doc/build/build/.images/color
+    - cp doc/build/xml/book.xml /tmp/kiwi-suse-doc/doc/build/xml
+    - git config --global user.email "kiwi-internal@suse.de"
+    - git config --global user.name "SUSE KIWI Team"
+    - cd /tmp/kiwi-suse-doc
+    - git add .
+    - git commit -a -m "upstream sync" && git push || true
```

Comment:
Probably a matter of taste, but... could you move that part into a separate shell script? IMHO, the GitLab Yaml file should be as simple as possible.